lldb/bindings/interface/SBValueDocstrings.i

```diff
@@ -45,6 +45,36 @@ SBValue instance which interprets the value object as representing the head of a
 linked list."
 ) lldb::SBValue;
 
+%feature("docstring", "
+Returns true if the SBValue holds any useful state
+and false otherwise.
+IsValid is a very weak API, lldb will only return
+invalid SBValues if it has no useful information
+about the SBValue.
+The two main ways you will end up with an invalid
+SBValue are:
+1) default constructed SBValues are not valid.
+2) SBValues that have outlived their SBTarget are
+no longer valid since its not safe to ask them
```

Comment:
What does it mean for it to be "not safe" to ask it questions? Does LLDB crash? Or does it modify state in an unpredictable manner? I think this should either be more specific about the consequences (e.g. "... no longer valid. Request data from the SBValue {is undefined, will crash LLDB, will return invalid data}.")

If you want to be vague about the behavior of this scenario (so that we aren't on the hook for guaranteeing the behavior stay the same) we should say that it is "undefined".

Reply:
See if you like this version better.

Reply:
I don't think the update addressed my comment. `GetError` will certainly tell you why something is invalid, but it doesn't specify the consequences of doing the unsafe action of asking the invalid SBValue questions. I guess what I'm asking is, does "not safe" refer to "safety against crashes" or "safety against invalid data"? I assume it's the latter, so I would suggest saying something like "It's not safe to ask them questions, the returned values may look valid even when they are not".

Reply:
GetError won't tell you why something is Invalid.  If it is invalid, it will just return a default constructed SBError, which has nothing useful in it.

What I intended was: my code is asking GetType, GetError, etc of an SBValue and everything it gets back is just an empty default-constructed return.  If you want to know why that might be, check IsValid, and if it's false, that's why.

Reply:
But if you are getting empty returns for the API's you are calling (GetType, etc) but the SBValue is valid, then there should be something useful in GetError.  If there's not, that's on us.

Reply:
If you just get back invalid/empty return values for the APIs you call on SBValue for invalid SBValues, why is it not safe to ask them questions? What does "safe" mean here?

Reply:
Note that I changed the phrasing to say "it WOULD not be safe.  Instead, lldb will return default constructed..."  That to me says "it isn't safe for lldb to ask questions of objects that depend on other objects that have been deleted, so we don't"...  I don't think I need to explain why it wouldn't be safe for lldb to ask these questions, though?  That seems obvious.

Reply:
I think I understand what you're saying better after re-reading it a few times. I formed a different meaning conceptually the first time I read it.

The two scenarios both have the exact same behavior. Why talk about safety in the second scenario but not the first? Rewording it a bit, it sounds like this: "Since it is not safe to ask them questions, SBValues that have outlived their SBTargets are invalid." Once the SBTarget dies and the SBValue is invalidated, it should behave exactly like the default constructed.

I don't understand the use of "safe" here. Safety implies that a risk exists, and I just don't see what could go wrong. The worst thing that could happen is you get back an invalid/empty object, and that's no different from asking a default constructed SBValue questions. LLDB shouldn't be crashing, shouldn't be silently corrupting or mutating state behind your back, and shouldn't be giving you valid-looking values that aren't valid when you ask a question of an invalid SBValue.

The second sentence didn't feel like it followed naturally from the first because of that. It describes something I would consider to be innocuous, so it didn't immediately click for me. Also, the subject of the sentence being lldb is too general. SBValue makes more sense since the target audience is developers trying to use SBValue. I was also confused at the use of "instead". I initially tried to interpret the sentence as describing an alternative to asking the questions, not an alternative to a behavior where it returned something other than a default constructed object.

My suggestion would be to write something like this:
"Calling methods on an invalid SBValue will return default constructed objects.
The two main ways you will end up with an invalid SBValue are:
1.) default constructed SBValues are not valid.
2.) SBValues that have outlived their SBTarget are
 no longer valid.
 
You can distinguish between an invalid SBValue and a valid SBValue with some missing data using IsValid when an SBValue returns a default constructed object from one of its methods."